File: 장세훈/Programmers/Level2/다음 큰 숫자.java
/*
    다음 큰 숫자
    https://school.programmers.co.kr/learn/courses/30/lessons/12911

    n의 2진수의 1의 개수 구하기 => 2로 나머지 연산해서 1이 나오면 카운팅
    Integer.bitCount(n) 메서드를 사용해서 1의 개수를 간단하게 구할 수도 있음.

*/

class Solution {
    public int solution(int n) {
        int nOneCount = countOne(n);
        for (int i = n + 1; ; i++) {
            if (nOneCount == countOne(i))
                return i;
        }
    }
    
public int countOne(int n) {
    int count = 0;

    while (n > 0) {
        if (n % 2 == 1)
            count++;
        n /= 2;
    }
    
    return count;
}

File: 정수빈/Programmers/Level2/다음큰숫자.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12911
# 2진수로 표현했을 때 1의 개수가 같은 숫자를 구하는 문제입니다. 
# 2진수로 변환한 후 1의 개수를 세어주었습니다.

def solution(n):
    answer = 0
    one = str(format(n, "b")).count("1")
    n2 = n+1
    while True:
        other = str(format(n2, "b")).count("1")
        if other == one:
            answer = n2
            break
        n2 += 1
    return answer